store/listings/views.py

```diff
@@ -1,3 +1,58 @@
-from django.shortcuts import render
+from typing import Dict
 
-# Create your views here.
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import serializers, viewsets
+
+from .models import Listing
+
+
+def swagger_docs(methods: Dict):
+    def decorator(cls):
+        for method, kwargs in methods.items():
+            original_method = getattr(cls, method, None)
+            if original_method:
+                decorated_method = swagger_auto_schema(**kwargs)(
+                    original_method)
+                setattr(cls, method, decorated_method)
+        return cls
+
+    return decorator
+
+
+class ListingSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Listing
+        fields = [
+            'id',
+            'title',
+            'price',
+            'currency_code',
+            'description',
+            'tags',
+            'created_on',
+            'updated_on'
+        ]
+
+
+listing_methods = {
+    'list': {
+        'operation_summary': "List all listings",
+        'operation_description': "Retrieve a list"
+                                 " of all listings in the database",
+        'tags': ["Listings"],
+        'responses': {200: ListingSerializer(many=True)}
+    },
+    'retrieve': {
+        'operation_summary': "Retrieve a listing",
+        'operation_description': "Retrieve a specific listing by ID",
+        'tags': ["Listings"],
+        'responses': {200: ListingSerializer()}
+    }
+}
```

Comment:
Need to hide other methods, otherwise swagger docs are weird:

Use my approach:

```python
listing_methods = {
    'list': {
        'operation_summary': "List all listings",
        'operation_description': "Retrieve a list"
                                 " of all listings in the database",
        'tags': ["Listings"],
        'responses': {200: ListingSerializer(many=True)}
    },
    'retrieve': {
        'operation_summary': "Retrieve a listing",
        'operation_description': "Retrieve a specific listing by ID",
        'tags': ["Listings"],
        'responses': {200: ListingSerializer()}
    },
    'create': {
        'auto_schema': None  # Use to hide from docs
    },
    'update': {
        'auto_schema': None # Use to hide from docs
    },
    'partial_update': {
        'auto_schema': None # Use to hide from docs
    },
    'destroy': {
        'auto_schema': None # Use to hide from docs
    }
}
```
![123](https://github.com/user-attachments/assets/3df80925-073a-4975-a7f1-9a28071db48c)

Reply:
Approach applied